### PR TITLE
Do not allow deploying in the root directory.

### PR DESCRIPTION
**Pull request by mitchell-as**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-1289" title="DX-1289" target="_blank">BugDX-1289</a>  State Deploy Failure On Ubuntu
  </td></table>
  <br />

### CODE REVIEW

**Review by MDrakos — APPROVED** [review]
Looks fine to me but you may want to get Nathan and/or Pete to take a look at the error message

**Review by Naatan — CHANGES_REQUESTED** [review]
The person encountering the bug did not seem to have ran deploy with a custom target path, meaning that something in our implementation defaulted to the root. I suspect this might be a case of the `HOME` env var not being set, but worth digging into that.

**Comment by mitchell-as** [comment]
@Naatan This is what happens in the docker file mentioned in the ticket. The command is run at root (/) so the deployment path was '.', which is root (/).

**Comment by Naatan** [comment]
@mitchell-as I see.. in that case can we just move the logic to here:

https://github.com/ActiveState/cli/blob/6adb01cf7a668ad1797e35108ea24b1b5e0794de/pkg/platform/runtime/setup/setup.go#L166-L166

That way it'll apply to all runtime usage. I don't think we're ever going to want to install to root.

**Comment by Naatan** [comment]
Also I think we need to resolve the path, as right now we wouldn't catch something like `/` or `../` from `/opt`.

**Comment by mitchell-as** [comment]
We are resolving it. `NewCustomTarget()` is calling `fileutils.ResolveUniqPath()`, which does what you are expecting. That is why I was comfortable with `== "/"`.

**Comment by mitchell-as** [comment]
Okay, moved the check into setup. Not sure about the wording though. Feel free to suggest something better.

File: pkg/platform/runtime/setup/setup.go [hunk 1]
```diff
@@ -163,6 +164,13 @@ func NewWithModel(target Targeter, msgHandler Events, model ModelProvider, an an
 
 // Update installs the runtime locally (or updates it if it's already partially installed)
 func (s *Setup) Update() error {
+	// Do not allow users to deploy runtimes to the root directory (this can easily happen in docker
+	// images). Note that runtime targets are fully resolved via fileutils.ResolveUniquePath(), so
+	// paths like "/." and "/opt/.." resolve to simply "/" at this time.
+	if (rt.GOOS != "windows" && s.target.Dir() == "/") {
+		return locale.NewInputError("err_runtime_setup_root", "Cannot set up a runtime in the root directory. Please try in another directory.")
```
Comment (Naatan) [inline pkg/platform/runtime/setup/setup.go:171]:
```suggestion
		return locale.NewInputError("err_runtime_setup_root", "Cannot set up a runtime in the root directory. Please specify or run from a user-writable directory.")
```

**Review by Naatan — CHANGES_REQUESTED** [review]

**Review by Naatan — APPROVED** [review]

**Pull request merged**